core/src/OpenACC/Kokkos_OpenACC_Macros.hpp

```diff
@@ -0,0 +1,23 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_OPENACC_MACROS_HPP
+#define KOKKOS_OPENACC_MACROS_HPP
+
+#define KOKKOS_IMPL_ACC_PRAGMA_HELPER(x) _Pragma(#x)
+#define KOKKOS_IMPL_ACC_PRAGMA(x) KOKKOS_IMPL_ACC_PRAGMA_HELPER(acc x)
```

Comment:
Haven't put much thought into the name.  Let me know if you want to rename to `KOKKOS_IMPL_PRAGMA_ACC` or if you have other suggestions.

Reply:
What's the reason for introducing these macros? You could use `_Pragma("bla")` instead of `#pragma bla` directly (also in macros), right?

Reply:
No I think you need an extra indirection to expand the operator.

Reply:
I suppose I don't actually need two macros if we spell out the `acc`
https://godbolt.org/z/j1GGd148G